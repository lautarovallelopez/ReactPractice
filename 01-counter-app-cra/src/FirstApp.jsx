import React from 'react';
import PropTypes from 'prop-types';

const greetingObject = {
    name: 'lautaro',
    greeting: 'bienvenido'
}
const greet = greeting => `${greeting.greeting} ${greeting.name}`;

const FirstApp = ({ title }) => {
    return (
        <>
            <h2>{greet(greetingObject)}</h2>
            <h3>Valle LÃ³pez</h3>
            <h4>{title}</h4>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
    title: 'Este es un titulo por defecto'
}

export default FirstApp;