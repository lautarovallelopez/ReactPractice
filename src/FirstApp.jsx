import React from 'react';
import PropTypes from 'prop-types';

const greetingObject = {
    name: 'lautaro',
    greeting: 'bienvenido'
}
const greet = greeting => `${greeting.greeting} ${greeting.name}`;

const FirstApp = ({ title }) => {
    return (
        <>
            <h2>{greet(greetingObject)}</h2>
            <h3>Valle LÃ³pez</h3>
            <h4 data-testid="test-title">{title}</h4>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired
};

export default FirstApp;