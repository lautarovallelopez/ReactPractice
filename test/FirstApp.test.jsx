import { render } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe('Prueba en <FirstApp />', () => {
    test('Debe hacer match con el snapshot', () => {
        const title = 'Este es mi titulo';
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el tÃ­tulo en un h4', () => {
        const title = 'Este es mi titulo';
        const { container, getByText } = render(<FirstApp title={title} />);
        expect(getByText(title)).toBeTruthy();

        const h4 = container.querySelector('h4');
        expect(h4.innerHTML).toBe(title);
    });
});