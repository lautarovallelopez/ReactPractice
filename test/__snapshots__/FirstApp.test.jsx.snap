// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h2>
    bienvenido lautaro
  </h2>
  <h3>
    Valle LÃ³pez
  </h3>
  <h4>
    Este es mi titulo
  </h4>
</div>
`;
